pylint/extensions/docparams.py

```diff
@@ -333,8 +333,7 @@ def visit_return(self, node: nodes.Return) -> None:
             return
 
         func_node = node.frame(future=True)
-        if not isinstance(func_node, astroid.FunctionDef):
```

Comment:
I'm pretty sure these circumstances are impossible, but I could be wrong.

Reply:
Could we get away with adding `type: ignore` instead of these assertions? They will show up as useless when `astroid` gets typing at some point.

Reply:
I replaced the assertions with type annotations. It seems to work.